servers/src/mining/stratumserver.rs

```diff
@@ -574,16 +575,21 @@ impl Handler {
 					// set a new deadline for rebuilding with fresh transactions
 					deadline = Utc::now().timestamp() + config.attempt_time_per_block as i64;
 
-					self.workers.update_block_height(new_block.header.height);
-					self.workers
-						.update_network_difficulty(state.current_difficulty);
-
+					// Manage list of versions of this heights block
 					if clear_blocks {
```

Comment:
A bit out of the scope but this would be probably be nicer to have:
```suggestion
					// If this is a new block, clear the current_block version history
					if current_hash != latest_hash {
```

Reply:
At that point in the code "current_hash" has been updated (see line 571)
`current_hash = latest_hash;`
So the check needs to be done before that.